[Feature] improve concat filter

aohsato opened the pull request:
## Status
 DEVELOPMENT

## Description
I increased the supported number of lidars from 2 to 8. autowarefoundation/autoware_ai#143
This limitation is by the template definition of message_filters Approximate Time Policy.
https://github.com/ros/ros_comm/blob/melodic-devel/utilities/message_filters/include/message_filters/sync_policies/approximate_time.h

## Related PRs
autowarefoundation/autoware#842

## Todos
- [x] Tests

## Steps to Test or Reproduce
![screenshot from 2018-05-10 02-48-23](https://user-images.githubusercontent.com/1531482/39831351-209096bc-53ff-11e8-9768-bfe1a9524862.png)

1. input topic names as list value to `Input Topics` field, like `[/points_alpha,/points_beta]`.
1. input `Output Topic` and `Output Frame`
1. execute `points_concat_filter`

aohsato committed:
Modify points_concat_filter to support up to 8 lidars

aohsato committed:
apply clang-format

aohsato committed:
Modify runtime_manager

amc-nu reviewed (changes requested):
Great work As usual.
Can you please check my comments?. The important one is the missing package. The other ones are optional.

yk-fujii committed:
apply clang-format

yk-fujii committed:
fix review comments
- fix CMakeFiles.txt and package.xml related yaml-cpp
- use input_topics_size_
- add brackets

yk-fujii committed:
Merge branch 'develop' of https://github.com/cpfl/autoware into feature/improve_concat_filter

aohsato committed:
Fix assertion condition

aohsato committed:
Fix cmake and remove msg from runtime manager

aohsato committed:
Remove yaml-cpp find package

aohsato committed:
Add -lyaml-cpp

aohsato committed:
Fix finding yaml-cpp by pkg_check_modules

aohsato committed:
Replaced yaml-cpp library flag
